src/components/Whiteboard/ElementBehaviors/Resizable/ResizeHandle.tsx

```diff
@@ -16,102 +16,150 @@
 
 import { Dispatch, RefObject, useCallback, useRef } from 'react';
 import { DraggableCore, DraggableData, DraggableEvent } from 'react-draggable';
+import { ElementKind } from '../../../../state';
 import { useSvgCanvasContext } from '../../SvgCanvas';
-import { ResizeHandlePosition } from './types';
-
-export function calculateResizeHandlePosition(
-  position: ResizeHandlePosition,
-  containerWidth: number,
-  containerHeight: number,
-  scale: number,
-): {
+import {
+  LineElementResizeHandlePosition,
+  PolylineAndShapeElementsResizeHandlePosition,
+  ResizeHandlePosition,
+} from './types';
+
+export type BaseParams = {
+  scale: number;
+};
+
+export type LineElementParams = BaseParams & {
+  elementKind: 'line';
+  handlePosition: LineElementResizeHandlePosition;
+  handlePositionX?: number;
+  handlePositionY?: number;
+};
+
+export type PolylineAndShapeElementsParams = BaseParams & {
+  elementKind: Exclude<ElementKind, 'line'>;
+  handlePosition: PolylineAndShapeElementsResizeHandlePosition;
+  containerWidth?: number;
+  containerHeight?: number;
+};
```

Comment:
`containerWidth` and `containerHeight` are defined for non lines, so they should not be optional. There is no reason to have separate types for different categories of element kinds if we don't specify explicitly what parameters are defined for each kind.

In order to make them not optional you would need to refactor usages of `calculateResizeHandlePosition` up to `ResizeHandleWrapper`